anesthetic/plot.py

```diff
@@ -816,16 +816,10 @@ def kde_contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
         data_y = data_y[weights != 0]
         weights = weights[weights != 0]
 
+    kde = gaussian_kde([data_x, data_y], weights=weights)
     cov = np.cov(data_x, data_y, aweights=weights)
     tri, w = triangular_sample_compression_2d(data_x, data_y, cov,
                                               weights, ncompress)
-    kde = gaussian_kde([tri.x, tri.y], weights=w)
-
-    x, y = kde.resample(ncompress)
-    x = np.concatenate([tri.x, x])
-    y = np.concatenate([tri.y, y])
-    w = np.concatenate([w, np.zeros(ncompress)])
-    tri = scaled_triangulation(x, y, cov)
```

Comment:
Was it ok to remove these lines? Tests don't complain...

Reply:
I think with the new setup whatever density problem this was trying to resolve is now resolved, and it certainly seems to be doing a better job with fewer lines

Reply:
Actually, these lines _do_ serve a purpose as the following example shows:

```python
data = np.random.randn(10000, 2)
data[:, 1] = data[:, 0]**2
ns = NestedSamples(data=data, columns=['x', 'y'])

fig = plt.figure()
fig, axes = make_2d_axes(['x', 'y'], fig=fig)
fig, axes = ns.plot_2d(axes)
```

### Without those lines:
![image](https://user-images.githubusercontent.com/18258042/178079080-769f1971-af4c-4147-b9f0-eb40a45a8887.png)


### With those lines:
![image](https://user-images.githubusercontent.com/18258042/178079129-89546602-418f-46b2-b79f-77d55b0d859d.png)

Reply:
Best reinstate them then unless you have another idea.